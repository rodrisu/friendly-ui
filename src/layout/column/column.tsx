import React from 'react'
import cc from 'classcat'

export type ColumnProps = Readonly<{
  className?: string
  children?: React.ReactNode
  key?: string
}>

/**
 * A child component for <Columns> parent component.
 * See <Columns> component for details.
 */
export const Column = (props: ColumnProps) => {
  const { className, children, key } = props

  return (
    <li key={key} className={cc(['kirk-column', className])}>
      {children}
    </li>
  )
}
