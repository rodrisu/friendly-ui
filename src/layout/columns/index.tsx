import styled from 'styled-components'

import { responsiveBreakpoints } from '../../_utils/branding'
import { Columns } from './columns'

const StyledColumns = styled(Columns)`
  & {
    display: flex;
    flex-direction: row;

    /** Reset styles from <ul> */
    list-style: none;
    padding: 0;
    margin: 0;
  }

  @media (${responsiveBreakpoints.isMediaSmall}) {
    & {
      flex-direction: column;
    }
  }
`

export { ColumnsProps } from './columns'
export { StyledColumns as Columns }
