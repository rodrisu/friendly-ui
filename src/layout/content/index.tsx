import { MainContentProps } from './Content'
import { BottomContent, Content, StyledMainContent } from './Content.style'

export { StyledMainContent as MainContent, BottomContent, Content, MainContentProps }
