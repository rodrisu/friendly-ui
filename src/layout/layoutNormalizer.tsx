import React from 'react'
import { createGlobalStyle } from 'styled-components'

import { horizontalSpace } from '../_utils/branding'

const horizontalPadding = (props: any): string => {
  if (props.noHorizontalSpacing) {
    return '0'
  }
  return props.hasHorizontalSpacing ? horizontalSpace.inner : horizontalSpace.global
}

const horizontalMargin = (props: any): string | number =>
  props.hasHorizontalSpacing ? horizontalSpace.outer : 0

export type NormalizeProps = Readonly<{
  hasHorizontalSpacing?: boolean
  noHorizontalSpacing?: boolean
}>

/**
 * Util method to normalize horizontal spacing
 * using !important because this should never be overridden
 */
export const normalizeHorizontally = (props: NormalizeProps): string => `
  padding-left: ${horizontalPadding(props)} !important;
  padding-right: ${horizontalPadding(props)} !important;
  margin-left: ${horizontalMargin(props)};
  margin-right: ${horizontalMargin(props)};
  `

// Legacy layout rules from production BBC.
const LegacyLayoutNormalizationGlobalStyles = createGlobalStyle`
    .home-column .kirk-item.kirk-item--clickable,
    .user-menu-item.kirk-item.kirk-item--clickable {
        margin-left: 0;
        margin-right: 0;
    }

    .home-column button.kirk-item.kirk-item--clickable,
    button.user-menu-item.kirk-item.kirk-item--clickable {
        max-width: 100%;
    }
`

const LayoutNormalizationGlobalStyles = createGlobalStyle`
  // Add normalization styles.
`

export type LayoutNormalizerProps = Readonly<{
  useLegacyNormalization?: boolean
}>

export const LayoutNormalizer = ({ useLegacyNormalization = true }: LayoutNormalizerProps) => {
  if (useLegacyNormalization) {
    return <LegacyLayoutNormalizationGlobalStyles />
  }

  return <LayoutNormalizationGlobalStyles />
}
