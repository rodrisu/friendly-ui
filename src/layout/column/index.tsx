import styled from 'styled-components'

import { Column } from './column'

const StyledColumn = styled(Column)`
  & {
    flex: 1;
    text-indent: 0;
    margin: 0;
    padding: 0;
  }
`

export { ColumnProps } from './column'
export { StyledColumn as Column }
