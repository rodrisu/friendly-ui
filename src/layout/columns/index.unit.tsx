import React from 'react'
import renderer from 'react-test-renderer'

import { Column } from '../../layout/column'
import { Columns } from './index'

describe('Columns', () => {
  it('should render default columns', () => {
    const columns = (
      <Columns>
        <Column>column 1</Column>
        <Column>column 2</Column>
        <Column>column 3</Column>
      </Columns>
    )
    const renderedColumns = renderer.create(columns).toJSON()
    expect(renderedColumns).toMatchSnapshot()
  })
})
