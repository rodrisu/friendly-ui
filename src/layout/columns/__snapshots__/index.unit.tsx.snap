// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Columns should render default columns 1`] = `
.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  text-indent: 0;
  margin: 0;
  padding: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  list-style: none;
  padding: 0;
  margin: 0;
}

@media (max-width:799px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<ul
  className="c0"
>
  <li
    className="kirk-column c1"
  >
    column 1
  </li>
  <li
    className="kirk-column c1"
  >
    column 2
  </li>
  <li
    className="kirk-column c1"
  >
    column 3
  </li>
</ul>
`;
